tensorflow_datasets/core/utils/tf_utils.py

```diff
@@ -135,6 +136,7 @@ def is_dtype(value):
   return True
 
 
+@functools.cache
```

Comment:
lru_cache as we're still supporting python 3.6

Reply:
done.